### ISSUE TITLE
Getting HTTP 404 on deployment

### ISSUE THREAD

**Issue by PedroC88** [tobania/VSTS.Extension.SqlReportingServices#53]
1. I have a self-hosted agent for VSTS
2. I can remote to this agent using the same credentials under which the VSTS agent service runs
3. While logged in I can open a browser and go to _http(s)://server/endpoint/ReportService2010.asmx?wsdl_
4. That URL returns an XML
5. I then copy-paste that URL into the _Webservice URL_ box in VSTS and provide the credentials
6. The release task fails with the following message: _The request failed with HTTP status 404: Not Found_.

Not sure what I'm doing wrong to configure the endpoint.

P.S. Before that error I also get the following message (not sure if related)
_System.Management.Automation.MethodInvocationException: Exception calling "GetFullPath" with "1" argument(s): "Illegal characters in path." ---> System.ArgumentException: Illegal characters in path._

P.S.S. nor the agent nor the reporting server are publicly available on the web, both computers are part of a private network.

**Comment by ygeelen-tb** [comment]
Pedro,

I think the GetFullPath error is your real issue here.
Can you please double check the parameter Remote Resource Path on existence?

And if that doesn't help, doublme check the parameters Local Report Path and Local Resource Path.

Otherwise, please provide screenshots of your full configuration.

Kind regards

**Comment by PedroC88** [comment]
**TL;DR** Switched to the right task, the _GetFullPath_ error doesn't appear anymore but I still get the 404

__
Based on the details that you asked for, I realized I was using the wrong task _SQL Server Reports Folder Deployment_ instead of _SQL Server Reports Deployment_. I now switched to the correct task which doesn't give me the _GetFullPath_ error but I still get the 404. Here are the variables 

1. **Local Report Path:** $(System.DefaultWorkingDirectory)/_Build Reports - QA/drop/reports/All Territories/*.rdl
2. **Report Upload Path:** _General Reports_ 
3. **Web Service URL:** _http(s)://server-name.domain.com/reportserver/ReportService2010.asmx?wsdl_

**Comment by ygeelen-tb** [comment]
Could you try with Remote Resource Path "/General Reports"?

And try to resolve the URL `http://server-name.domain.com/ReportServer?%2fGeneral+Reports&rs:Command=ListChildren`

**Comment by ygeelen-tb** [comment]
`http://server-name.domain.com/ReportServer?%2fGeneral+Reports&rs:Command=ListChildren`

**Comment by PedroC88** [comment]
1. Using _/General Reports_ didn't make any difference
2. Browing to `http://server-name.domain.com/ReportServer?%2fGeneral+Reports&rs:Command=ListChildren` lists all reports within the folder

**Comment by ygeelen-tb** [comment]
https://github.com/tobania/VSTS.Extension.SqlReportingServices/blob/master/SqlReportingServices/SqlReportingServicesDeployment/SqlReportingServicesDeployment.ps1

Could you enable Verbose logging at the bottom of the task and compare the logging output to the Write statements in the script here?
That could give an indication how far you're getting and which param could be the problem.

**Comment by PedroC88** [comment]
The last message I get before the error is on line 118
> Creating WebService proxy using credentials

![image](https://user-images.githubusercontent.com/6889567/52791372-2e566e80-303f-11e9-9333-d9a4098e9ca9.png)

The first error line continues as
> The request failed with HTTP status 404: Not Found.

**Comment by PedroC88** [comment]
To debug the issue I extracted and simplified the section _Creating Webservice proxy_. I then ran the script in my computer and in the agent. In my computer it failed with the same 404, in the agent it worked. The 404 is expected in my computer because I don't have access to that SSRS server. Success is also expected from the agent as it has access to the SSRS server. Here's the script I used.

```
$WsUsername = "user.name";
$wsSecurePass = ConvertTo-SecureString -String "p4ssw0rd" -AsPlainText -Force
$WebserviceUrl = "http(s)://server-name.domain.com/reportserver/ReportService2010.asmx?wsdl"
[System.Management.Automation.PSCredential] $auth = New-Object System.Management.Automation.PSCredential -ArgumentList $WsUsername,$wsSecurePass;
$ssrs = New-WebServiceProxy -Uri $WebserviceUrl -Credential $auth -ErrorAction Stop;
```

Could it be that somehow the PS script is not running on the agent computer (or perhaps under some different credentials)?

**P.S.** Your script has a small _bug_ where you declare a variable called $type for the SSRS service, but when you print it for reference is called $dataType which of course is empty.

**Comment by ygeelen-tb** [comment]
Thanks for checking those things. I would advise to check 2 more things:

- The Run on agent task in your release definition: which agent is selected?
- In your Windows Services you should see a service for your agent. Can you double check the user that's running that service has access to your SSRS server?

**Comment by PedroC88** [comment]
OMG... I'm embarrassed I missed something so obvious as the agent (specially after asking the question myself, [facepalm]). That was the issue, it was running under the hosted agent which of course doesn't have access to the reporting server.

I switched it to the proper pool and was able to move forward, some reports appear to be failing but it seems an issue on the definition of the reports so that's on me.

Thank you so much @ygeelen-tb!!

**Comment by ygeelen-tb** [comment]
@PedroC88 No worries I see people forgetting about that regularly! They should better display the chosen agent in the task title so it's less easy to miss.
